sstables_loader.cc

```diff
@@ -190,6 +195,21 @@ class tablet_sstable_streamer : public sstable_streamer {
 };
 
 inet_address_vector_replica_set sstable_streamer::get_endpoints(const dht::token& token) const {
+    return get_all_endpoints(token) | std::views::filter([&topo = _erm->get_topology(), scope = _stream_scope] (const auto& ep) {
+        switch (scope) {
+        case stream_scope::all:
+            return true;
+        case stream_scope::dc:
+            return topo.get_datacenter(ep) == topo.get_datacenter();
+        case stream_scope::rack:
+            return topo.get_rack(ep) == topo.get_rack();
+        case stream_scope::node:
+            return topo.is_me(ep);
```

Comment:
I would drop the `node` scope initially as it is already achievable by omitting the --load-and-stream option entirely and it's unclear how to combine it with --primary-replica-only.
Also, AFAIK it is not a requirement for fast restore, that is likely to use rack scope, and maybe dc.

Reply:
The node scope is slightly different in the way it's effectively implemented.
Omitting --load-and-stream will load sstables locally and run dedicated reshard and reshape jobs on them
Using scope=node will load sstables locally and will read, merge and stream mutation to self distributing them with write_mutations_on_all_shards.
I would look into replacing the former implementation with the latter, as it's less code to maintain.

Regarding fast restore. I think we don't yet have enough information to decide. Consider you need to restore sstable S into nodes A, B and C (same dc and rack). There are two options:
1. A loads S then streams to B and C
2. A, B and C all load S with node scope on their own
So we trade 2x intra-node transfers to s3-to-node ones. Which one is faster? What if there are N sstables to so the trade is 2Nx intra-node vs 3s-to-node?